test-data/unit/check-optional.test

```diff
@@ -162,6 +163,15 @@ x = None
 x = 1
 reveal_type(x)  # E: Revealed type is 'builtins.int'
 
+[case testInferOptionalAnyType]
+from typing import Any
+x = None
+a = None  # type: Any
```

Comment:
Is this issue specific to ``Union[None, Any]`` or to arbitrary declared optional type? In the latter case I would another test case with ``int`` instead of ``Any`` in the declared type.